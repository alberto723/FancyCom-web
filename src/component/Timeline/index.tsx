import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
// import './style.css';

interface dataProps {
  title: string;
  subTitle: string;
  date: string;
  icon: object;
}

export interface TimelineProps {
  type?: string;
  data?: dataProps[];
  // style?: ViewStyle;
  barColor?: string;
  // barStyle?: ViewStyle;
  lineColor?: string;
  // lineStyle?: ViewStyle;
}

export const Timeline: React.FC<TimelineProps> = ({
  type,
  data,
  // style,
  barColor,
  // barStyle,
  lineColor
  // lineStyle
}) => {
  const styles: any = useStyles();
  const renderTimeline = () => {
    return (
      <>
        {data &&
          data.map((item, index) => (
            <div key={index} className={styles.scheduleItemContainer}>
              <div className={styles.barSection}>
                <div className={styles.iconCircle} style={{ backgroundColor: barColor }}>
                  {item.icon}
                </div>
                <div className={styles.timelineBar} style={{ backgroundColor: barColor }} />
              </div>
              <div className={styles.descriptionSection}>
                <p className={styles.dateText}>{item.date}</p>
                <div className={styles.textPart}>
                  <p className={styles.title}>{item.title}</p>
                  <p className={styles.content}>{item.subTitle}</p>
                </div>
              </div>
            </div>
          ))}
      </>
    );
  };

  return (
    <div className={styles.screen}>
      <div className={styles.timelineContainer}>{type === 'VERTICAL' && renderTimeline()}</div>
    </div>
  );
};

Timeline.defaultProps = {
  type: 'VERTICAL'
};

const useStyles = makeStyles(() => ({
  screen: {
    flex: 1,
    backgroundColor: 'white',
    paddingTop: '15px',
  },
  timelineContainer: {
    padding: '15px',
  },
  scheduleItemContainer: {
    display: '-webkit-box',
    flexDirection: 'row',
  },
  barSection: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  iconCircle: {
    width: '28px',
    height: '28px',
    borderRadius: '14px',
    backgroundColor: '#49A6E9',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center'
  },
  timelineBar: {
    display: 'flex',
    flex: 1,
    width: '3px',
    height: '100%',
    backgroundColor: '#49A6E9',
  },
  descriptionSection: {
    display: 'flex',
    flexDirection: 'column',
    marginLeft: '7px',
  },
  dateText: {
    marginTop: '5px',
    marginBottom: '5px',
    color: 'darkgrey',
  },
  textPart: {
    borderColor: '#F1F1F1',
    borderWidth: '2px',
    borderStyle: 'solid',
    width: '100%',
    padding: '10px',
    marginBottom: '20px',
    marginTop: '5px',
  },
  title: {
    color: 'dimgrey',
    fontWeight: 'bold',
  },
  content: {
    marginTop: '5px',
    color: 'darkgrey',
  },
}));
