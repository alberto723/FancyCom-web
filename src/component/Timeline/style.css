.screen {
  flex: 1;
  background-color: white;
}
.timelineContainer {
  padding: 15px;
}
.scheduleItemContainer {
  display: -webkit-box;
  flex-direction: row;
}
.barSection {
  display: flex;
  flex-direction: column;
  align-items: center;
}
.iconCircle {
  width: 28px;
  height: 28px;
  border-radius: 14px;
  background-color: #49A6E9;
  display: flex;
  justify-content: center;
  align-items: center;
}
.timelineBar {
  display: flex;
  flex: 1;
  width: 3px;
  height: 100%;
  background-color: #49A6E9;
}
.descriptionSection {
  display: flex;
  flex-direction: column;
  margin-left: 7px;
}
.dateText {
  margin-top: 5px;
  margin-bottom: 5px;
  color: darkgrey;
}
.textPart {
  border-color: #F1F1F1;
  border-width: 2px;
  border-style: solid;
  width: 100%;
  padding: 10px;
  margin-bottom: 20px;
  margin-top: 5px;
}
.title {
  color: dimgrey;
  font-weight: bold;
}
.content {
  margin-top: 5px;
  color: darkgrey;
}
