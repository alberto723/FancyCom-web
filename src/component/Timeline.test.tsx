// Canvas.test.js
import { render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';

import { Timeline } from './Timeline';

import CheckIcon from '@material-ui/icons/Check';
import DraftsIcon from '@material-ui/icons/Drafts';
import EditIcon from '@material-ui/icons/Edit';
import LaunchIcon from '@material-ui/icons/Launch';
import LockIcon from '@material-ui/icons/Lock';
import NotificationsActiveIcon from '@material-ui/icons/NotificationsActive';

const timelineItems = [
  {
    title: 'Completed profile data',
    subTitle:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
    date: 'Jan 25, 2020',
    icon: <CheckIcon style={{ color: 'white', fontSize: 17 }} />
  },
  {
    title: 'Completed profile data',
    subTitle:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
    date: 'Jan 25, 2020',
    icon: <DraftsIcon style={{ color: 'white', fontSize: 17 }} />
  },
  {
    title: 'Completed profile data',
    subTitle:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
    date: 'Jan 25, 2020',
    icon: <EditIcon style={{ color: 'white', fontSize: 17 }} />
  },
  {
    title: 'Completed profile data',
    subTitle:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
    date: 'Jan 25, 2020',
    icon: <LaunchIcon style={{ color: 'white', fontSize: 17 }} />
  },
  {
    title: 'Completed profile data',
    subTitle:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
    date: 'Jan 25, 2020',
    icon: <LockIcon style={{ color: 'white', fontSize: 17 }} />
  },
  {
    title: 'Completed profile data',
    subTitle:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
    date: 'Jan 25, 2020',
    icon: <NotificationsActiveIcon style={{ color: 'white', fontSize: 17 }} />
  }
];

test('Should be rendered in the dom', () => {
  const component = renderer.create(
    <Timeline
      data={timelineItems}
      type='VERTICAL'
      barColor='#49A6E9'
      lineColor='#49A6E9'
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should show the same number of events as there are items in the data prop array', () => {
  const { queryByLabelText, getByText, container } = render(
    <Timeline
      data={timelineItems}
      type='VERTICAL'
      barColor='#49A6E9'
      lineColor='#49A6E9'
    />
  );

  expect(container.querySelectorAll('.makeStyles-scheduleItemContainer-3').length).toBe(timelineItems.length);
});

test('Should render each event with a title matching the value in its data array item', () => {
  const { queryByLabelText, getByText, container } = render(
    <Timeline
      data={timelineItems}
      type='VERTICAL'
      barColor='#49A6E9'
      lineColor='#49A6E9'
    />
  );

  expect(container.querySelector('.makeStyles-title-10')?.innerHTML).toBe('Completed profile data');
});

test('Should render the title for each event with a style matching the value in its data array item', () => {
  const { queryByLabelText, getByText, container } = render(
    <Timeline
      data={timelineItems}
      type='VERTICAL'
      barColor='#49A6E9'
      lineColor='#49A6E9'
    />
  );

  expect(container.querySelector('.makeStyles-title-10')?.style != null).toBe(true);
});

test('Should render each event with a subtitle matching the value in its data array item', () => {
  const { queryByLabelText, getByText, container } = render(
    <Timeline
      data={timelineItems}
      type='VERTICAL'
      barColor='#49A6E9'
      lineColor='#49A6E9'
    />
  );

  expect(container.querySelector('.makeStyles-content-11')?.innerHTML).toBe('Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.');
});

test('Should render the subtitle for each event with a style matching the value in its data array item', () => {
  const { queryByLabelText, getByText, container } = render(
    <Timeline
      data={timelineItems}
      type='VERTICAL'
      barColor='#49A6E9'
      lineColor='#49A6E9'
    />
  );

  expect(container.querySelector('.makeStyles-content-11')?.className != null).toBe(true);
});

test('Should render the start and end dates for each event positioned at a location matching the values in its data array item', () => {
  const { queryByLabelText, getByText, container } = render(
    <Timeline
      data={timelineItems}
      type='VERTICAL'
      barColor='#49A6E9'
      lineColor='#49A6E9'
    />
  );

  expect(container.querySelector('.makeStyles-dateText-8')?.innerHTML != null).toBe(true);
});

test('Should render an icon for each event that matches the value in its data array item', () => {
  const { queryByLabelText, getByText, container } = render(
    <Timeline
      data={timelineItems}
      type='VERTICAL'
      barColor='#49A6E9'
      lineColor='#49A6E9'
    />
  );

  expect(container.querySelector('.makeStyles-iconCircle-5')?.innerHTML != null).toBe(true);
});

test('Should render the icon for each event with a style matching the iconStyle value in its data array item', () => {
  const { queryByLabelText, getByText, container } = render(
    <Timeline
      data={timelineItems}
      type='VERTICAL'
      barColor='#49A6E9'
      lineColor='#49A6E9'
    />
  );

  expect(container.querySelector('.makeStyles-iconCircle-5')?.innerHTML != null).toBe(true);
});

test('Should render each activity event with the zIndex order provided in its data array item', () => {
  const { queryByLabelText, getByText, container } = render(
    <Timeline
      data={timelineItems}
      type='VERTICAL'
      barColor='#49A6E9'
      lineColor='#49A6E9'
    />
  );

  expect(container.querySelector('.makeStyles-iconCircle-5')?.innerHTML != null).toBe(true);
});

test('Should show the timeline using the style provided in the style props', () => {
  const { queryByLabelText, getByText, container } = render(
    <Timeline
      data={timelineItems}
      type='VERTICAL'
      barColor='#49A6E9'
      lineColor='#49A6E9'
    />
  );

  expect(container.querySelector('.makeStyles-scheduleItemContainer-3')?.className != null).toBe(true);
});

test('Should show the bars in the provided color', () => {
  const { queryByLabelText, getByText, container } = render(
    <Timeline
      data={timelineItems}
      type='VERTICAL'
      barColor='#49A6E9'
      lineColor='#49A6E9'
    />
  );

  expect(container.querySelector('.makeStyles-barSection-4').className != null).toBe(true);
});

test('Should show the bars using the style provided int he style props', () => {
  const { queryByLabelText, getByText, container } = render(
    <Timeline
      data={timelineItems}
      type='VERTICAL'
      barColor='#49A6E9'
      lineColor='#49A6E9'
    />
  );

  expect(container.querySelector('.makeStyles-timelineBar-6').className != null).toBe(true);
});

test('Should show the lines in the provided color', () => {
  const { queryByLabelText, getByText, container } = render(
    <Timeline
      data={timelineItems}
      type='VERTICAL'
      barColor='#49A6E9'
      lineColor='#49A6E9'
    />
  );

  expect(container.querySelector('.makeStyles-timelineBar-6').className != null).toBe(true);
});

test('Should show the lines using the style provided int he style props', () => {
  const { queryByLabelText, getByText, container } = render(
    <Timeline
      data={timelineItems}
      type='VERTICAL'
      barColor='#49A6E9'
      lineColor='#49A6E9'
    />
  );

  expect(container.querySelector('.makeStyles-timelineBar-6').className != null).toBe(true);
});
