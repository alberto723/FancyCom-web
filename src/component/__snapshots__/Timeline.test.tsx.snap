// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be rendered in the dom 1`] = `
<div
  className="makeStyles-screen-1"
>
  <div
    className="makeStyles-timelineContainer-2"
  >
    <div
      className="makeStyles-scheduleItemContainer-3"
    >
      <div
        className="makeStyles-barSection-4"
      >
        <div
          className="makeStyles-iconCircle-5"
          style={
            Object {
              "backgroundColor": "#49A6E9",
            }
          }
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "color": "white",
                "fontSize": 17,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
            />
          </svg>
        </div>
        <div
          className="makeStyles-timelineBar-6"
          style={
            Object {
              "backgroundColor": "#49A6E9",
            }
          }
        />
      </div>
      <div
        className="makeStyles-descriptionSection-7"
      >
        <p
          className="makeStyles-dateText-8"
        >
          Jan 25, 2020
        </p>
        <div
          className="makeStyles-textPart-9"
        >
          <p
            className="makeStyles-title-10"
          >
            Completed profile data
          </p>
          <p
            className="makeStyles-content-11"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </p>
        </div>
      </div>
    </div>
    <div
      className="makeStyles-scheduleItemContainer-3"
    >
      <div
        className="makeStyles-barSection-4"
      >
        <div
          className="makeStyles-iconCircle-5"
          style={
            Object {
              "backgroundColor": "#49A6E9",
            }
          }
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "color": "white",
                "fontSize": 17,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M21.99 8c0-.72-.37-1.35-.94-1.7L12 1 2.95 6.3C2.38 6.65 2 7.28 2 8v10c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2l-.01-10zM12 13L3.74 7.84 12 3l8.26 4.84L12 13z"
            />
          </svg>
        </div>
        <div
          className="makeStyles-timelineBar-6"
          style={
            Object {
              "backgroundColor": "#49A6E9",
            }
          }
        />
      </div>
      <div
        className="makeStyles-descriptionSection-7"
      >
        <p
          className="makeStyles-dateText-8"
        >
          Jan 25, 2020
        </p>
        <div
          className="makeStyles-textPart-9"
        >
          <p
            className="makeStyles-title-10"
          >
            Completed profile data
          </p>
          <p
            className="makeStyles-content-11"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </p>
        </div>
      </div>
    </div>
    <div
      className="makeStyles-scheduleItemContainer-3"
    >
      <div
        className="makeStyles-barSection-4"
      >
        <div
          className="makeStyles-iconCircle-5"
          style={
            Object {
              "backgroundColor": "#49A6E9",
            }
          }
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "color": "white",
                "fontSize": 17,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </div>
        <div
          className="makeStyles-timelineBar-6"
          style={
            Object {
              "backgroundColor": "#49A6E9",
            }
          }
        />
      </div>
      <div
        className="makeStyles-descriptionSection-7"
      >
        <p
          className="makeStyles-dateText-8"
        >
          Jan 25, 2020
        </p>
        <div
          className="makeStyles-textPart-9"
        >
          <p
            className="makeStyles-title-10"
          >
            Completed profile data
          </p>
          <p
            className="makeStyles-content-11"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </p>
        </div>
      </div>
    </div>
    <div
      className="makeStyles-scheduleItemContainer-3"
    >
      <div
        className="makeStyles-barSection-4"
      >
        <div
          className="makeStyles-iconCircle-5"
          style={
            Object {
              "backgroundColor": "#49A6E9",
            }
          }
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "color": "white",
                "fontSize": 17,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
          </svg>
        </div>
        <div
          className="makeStyles-timelineBar-6"
          style={
            Object {
              "backgroundColor": "#49A6E9",
            }
          }
        />
      </div>
      <div
        className="makeStyles-descriptionSection-7"
      >
        <p
          className="makeStyles-dateText-8"
        >
          Jan 25, 2020
        </p>
        <div
          className="makeStyles-textPart-9"
        >
          <p
            className="makeStyles-title-10"
          >
            Completed profile data
          </p>
          <p
            className="makeStyles-content-11"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </p>
        </div>
      </div>
    </div>
    <div
      className="makeStyles-scheduleItemContainer-3"
    >
      <div
        className="makeStyles-barSection-4"
      >
        <div
          className="makeStyles-iconCircle-5"
          style={
            Object {
              "backgroundColor": "#49A6E9",
            }
          }
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "color": "white",
                "fontSize": 17,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
            />
          </svg>
        </div>
        <div
          className="makeStyles-timelineBar-6"
          style={
            Object {
              "backgroundColor": "#49A6E9",
            }
          }
        />
      </div>
      <div
        className="makeStyles-descriptionSection-7"
      >
        <p
          className="makeStyles-dateText-8"
        >
          Jan 25, 2020
        </p>
        <div
          className="makeStyles-textPart-9"
        >
          <p
            className="makeStyles-title-10"
          >
            Completed profile data
          </p>
          <p
            className="makeStyles-content-11"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </p>
        </div>
      </div>
    </div>
    <div
      className="makeStyles-scheduleItemContainer-3"
    >
      <div
        className="makeStyles-barSection-4"
      >
        <div
          className="makeStyles-iconCircle-5"
          style={
            Object {
              "backgroundColor": "#49A6E9",
            }
          }
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "color": "white",
                "fontSize": 17,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M7.58 4.08L6.15 2.65C3.75 4.48 2.17 7.3 2.03 10.5h2c.15-2.65 1.51-4.97 3.55-6.42zm12.39 6.42h2c-.15-3.2-1.73-6.02-4.12-7.85l-1.42 1.43c2.02 1.45 3.39 3.77 3.54 6.42zM18 11c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2v-5zm-6 11c.14 0 .27-.01.4-.04.65-.14 1.18-.58 1.44-1.18.1-.24.15-.5.15-.78h-4c.01 1.1.9 2 2.01 2z"
            />
          </svg>
        </div>
        <div
          className="makeStyles-timelineBar-6"
          style={
            Object {
              "backgroundColor": "#49A6E9",
            }
          }
        />
      </div>
      <div
        className="makeStyles-descriptionSection-7"
      >
        <p
          className="makeStyles-dateText-8"
        >
          Jan 25, 2020
        </p>
        <div
          className="makeStyles-textPart-9"
        >
          <p
            className="makeStyles-title-10"
          >
            Completed profile data
          </p>
          <p
            className="makeStyles-content-11"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
